process/block/shardblock.go

```diff
@@ -506,12 +516,29 @@ func (sp *shardProcessor) indexBlockIfNeeded(
 	}
 
 	shardId := sp.shardCoordinator.SelfId()
-	pubKeys, err := sp.nodesCoordinator.GetValidatorsPublicKeys(header.GetPrevRandSeed(), header.GetRound(), shardId)
+
+	// TODO: remove if epoch start block needs to be validated by the new epoch nodes
+	epoch := header.GetEpoch()
+	if header.IsStartOfEpochBlock() && epoch > 0 {
+		epoch = epoch - 1
+	}
+
+	pubKeys, err := sp.nodesCoordinator.GetValidatorsPublicKeys(
```

Comment:
This method could be renamed to `GetValidatorsPublicKeysForConsensusGroup()`, for clarity.

Reply:
renamed to GetConsensusValidatorsPublicKeys